src/build_options.rs

```diff
@@ -560,6 +563,16 @@ impl BuildOptions {
         let generate_import_lib = is_generating_import_lib(&cargo_metadata)?;
         let interpreter = if self.find_interpreter {
             // Auto-detect interpreters
+            let requires_python = metadata23.requires_python.as_ref();
```

Comment:
How about moving this check after L588, and use it as a filter to [retain](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.retain) only non-prerelease versions? Something like this:

```rust
// existing code
let mut interpreter = if self.find_interpreter { ... } else { ... };
interpreter.retain(|interp| {
    // your added code here
});
```